src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -91,6 +92,21 @@ public class SocketOutput : ISocketOutput
                 if (buffer.Count > 0)
                 {
                     var tail = ProducingStart();
+                    if (tail.IsDefault)
+                    {
+                        if (cancellationToken.CanBeCanceled)
+                        {
+                            return TaskUtilities.GetCancelledTask(
+                                cancellationToken.IsCancellationRequested ?
+                                cancellationToken :
+                                new CancellationToken(true));
```

Comment:
We only want to throw if the token is actually canceled. This seems to be throwing in any case there is a cancelable token and the socket is closed.